pkg/infrastructure/gcp/clusterapi/iam.go

```diff
@@ -102,48 +105,57 @@ func AddServiceAccountRoles(ctx context.Context, projectID, serviceAccountID str
 		return fmt.Errorf("failed to create resourcemanager service: %w", err)
 	}
 
-	policy, err := getPolicy(ctx, service, projectID)
-	if err != nil {
-		return err
-	}
+	backoff := time.Second
+	for {
```

Comment:
Why not use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoffWithContext which also checks the context for cancellation/timeout in between tries?

Reply:
It doesn't need to be `ExponentialBackoff`. For example, there is also https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#UntilWithContext